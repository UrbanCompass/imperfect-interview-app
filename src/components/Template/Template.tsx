/* eslint-disable react/no-danger */
import React from 'react';

import { Title } from '@/components/common';
import { REACT_APP_PLACEHOLDER, UC_GLOBALS_ID } from '@/constants';

interface TemplateProps {
  analyticsUrl?: string;
  assets: Record<string, string>;
  ucGlobals: string;
  isLocal: boolean;
}

/* istanbul ignore next */
function Template({ assets, analyticsUrl = '', isLocal, ucGlobals }: TemplateProps): JSX.Element {
  return (
    <html lang="en-US">
      <head>
        <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        {analyticsUrl && <script src={analyticsUrl} />}
        <link rel="stylesheet" href="/ucfe-assets/normalize.css/8/normalize.min.css" />
        <link rel="stylesheet" href="/ucfe-assets/util-classes.css/2/util-classes.min.css" />
        <link rel="stylesheet" href="/ucfe-assets/fonts/3/fonts.min.css" />
        <link rel="stylesheet" href="/ucfe-assets/cx/2/cx.min.css" />
        <title>{isLocal && '[local] '}Video Generator</title>
      </head>
      <body className="uc-appBody">
        <div className="uc-appContainer">
          <script
            id={UC_GLOBALS_ID}
            dangerouslySetInnerHTML={{ __html: ucGlobals }}
            type="application/json"
          />
          <Title />
          <div id="root" css={{ width: '100%' }}>
            {REACT_APP_PLACEHOLDER}
          </div>
          <div id="portal-root" />
        </div>
        {assets['vendor.js'] && <script type="text/javascript" src={assets['vendor.js']} />}
        {assets['client.js'] && <script type="text/javascript" src={assets['client.js']} />}
      </body>
    </html>
  );
}

export { Template };
