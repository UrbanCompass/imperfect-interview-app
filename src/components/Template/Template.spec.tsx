import React from 'react';
import { shallow } from 'enzyme';

import { Template } from './Template';

describe('Template', (): void => {
  const props = {
    analyticsUrl: 'http://my.analytics',
    ucGlobals: 'something',
    assets: {
      'vendor.js': '/path/to/vendor.js',
      'client.js': '/path/to/client.js',
    },
    isLocal: false,
  };

  it('renders', (): void => {
    expect(shallow(<Template {...props} />)).toMatchSnapshot();
  });
});
