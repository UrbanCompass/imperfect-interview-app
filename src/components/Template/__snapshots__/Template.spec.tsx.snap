// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template renders 1`] = `
<html
  lang="en-US"
>
  <head>
    <meta
      content="IE=edge"
      httpEquiv="X-UA-Compatible"
    />
    <meta
      charSet="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <script
      src="http://my.analytics"
    />
    <link
      href="/ucfe-assets/normalize.css/8/normalize.min.css"
      rel="stylesheet"
    />
    <link
      href="/ucfe-assets/util-classes.css/2/util-classes.min.css"
      rel="stylesheet"
    />
    <link
      href="/ucfe-assets/fonts/3/fonts.min.css"
      rel="stylesheet"
    />
    <link
      href="/ucfe-assets/cx/2/cx.min.css"
      rel="stylesheet"
    />
    <title>
      Video Generator
    </title>
  </head>
  <body
    className="uc-appBody"
  >
    <div
      className="uc-appContainer"
    >
      <script
        dangerouslySetInnerHTML={
          Object {
            "__html": "something",
          }
        }
        id="listing-video-editor-uc-globals"
        type="application/json"
      />
      <Title />
      <div
        css={
          Object {
            "width": "100%",
          }
        }
        id="root"
      >
        REACT_APP_PLACEHOLDER
      </div>
      <div
        id="portal-root"
      />
    </div>
    <script
      src="/path/to/vendor.js"
      type="text/javascript"
    />
    <script
      src="/path/to/client.js"
      type="text/javascript"
    />
  </body>
</html>
`;
